unit_tests/test_grid_fee_calculation.py

```diff
@@ -137,14 +137,42 @@ def test_grid_fee_is_calculated_correctly_for_market_to_market(self, fee_type):
                 target_market_or_device_name=leaf_names[1],
                 fee_type=fee_type))
 
-    @parameterized.expand([['last_market_fee'], ['next_market_fee']])
+    @parameterized.expand(['last_market_fee', 'next_market_fee'])
     def test_grid_fee_is_calculated_correctly_for_device_to_parent(self, fee_type):
         if fee_type == 'next_market_fee':
             leaf_name_expected_fee = {3.1: ['House 2.1', 'Load 2.1'],
                                       2.2: ['Load 1.2', 'House 1.2', ]}
         else:
             leaf_name_expected_fee = {2.1: ['House 2.1', 'Load 2.1'],
-                                      1.2: ['Load 1.2', 'House 1.2', ]}
+                                      1.2: ['Load 1.2', 'House 1.2']}
+        for expected_fee, leaf_names in leaf_name_expected_fee.items():
+            assert isclose(expected_fee, self.grid_fee_calc.calculate_grid_fee(
+                start_market_or_device_name=leaf_names[0],
+                target_market_or_device_name=leaf_names[1],
+                fee_type=fee_type))
+
+    @parameterized.expand(['last_market_fee', 'next_market_fee'])
+    def test_grid_fee_is_calculated_correctly_for_device_to_street(self, fee_type):
```

Comment:
```suggestion
    def test_grid_fee_is_calculated_correctly_for_device_to_intermediate_area(self, fee_type):
```
Its better to avoid referring to a hard coded area name

Reply:
Hmm, I don't agree. This is a test for a specific case. The whole module depends on the representation in the top.
And here in this test case it is important that the target area is the street and not the house. `intermediate_area` is not a correct description for the street, IMO. If you could think of a more universal way of describing "two levels up", let me know and I will change it.

Reply:
@faizan2590 if you agree, I would push now.

Reply:
@hannesdiedrich Please go ahead then!